03.bowling/bowling.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+def main
+  scoreboard = ARGV[0]
+  frames = split_by_frames(scoreboard)
+  final_score = calculate_score(frames)
+  output_result(final_score)
+end
+
+def strike?(frame)
+  frame == [10, 0]
+end
+
+def spare?(frame)
+  !strike?(frame) && (frame[0] + frame[1] == 10)
+end
+
+def group_last_frame_shots(frames, last_frame, last_frame_added_shot)
+  # 10フレーム目が全てストライクの場合
+  if strike?(last_frame) && strike?(last_frame_added_shot)
+    last_frame_final_shot = frames[11]
+    (last_frame + last_frame_added_shot + last_frame_final_shot).reject(&:zero?)
+  else
+    (last_frame + last_frame_added_shot).reject(&:zero?)
+  end
+end
+
+def split_by_frames(scoreboard)
+  scores = scoreboard.split(',')
+
+  shots = scores.flat_map do |score|
+    score == 'X' ? [10, 0] : score.to_i
+  end
+
+  shots.each_slice(2).to_a
+end
+
+def detect_bonus_score(frames, frame, index)
+  next_frame = frames[index + 1]
+
+  if spare?(frame)
+    next_frame[0]
+  elsif strike?(frame)
+    if strike?(next_frame)
+      after_next_frame = frames[index + 2]
+      next_frame[0] + after_next_frame[0]
+    else
+      next_frame[0..1].sum
+    end
+  else
+    0
+  end
+end
+
+def calculate_score(frames)
+  final_score = 0
+
+  frames[0..8].each_with_index.sum do |frame, index|
```

Comment:
sumを使うならこうですね。
APIドキュメントをよく読んで使い方を確認してみてください〜。

```suggestion
  final_score = frames[0..8].each_with_index.sum do |frame, index|
```

Reply:
なるほど、[ドキュメントのsumメソッドの説明](https://docs.ruby-lang.org/ja/latest/method/Array/i/sum.html)で
> 要素の合計を返します。

と記載してあることからも、
- ループの中で自己代入しなくてよい
- `final_score = 0` を敢えて用意しなくてよい

と考えることができますね🫡

対応commit:[843a4ff](https://github.com/Judeeeee/ruby-practices/pull/5/commits/843a4ff94c2385a485e8f8ed29dd64e9167a2a3e)